packages/wasm/src/transform/mod.rs

```diff
@@ -0,0 +1,336 @@
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        program
+    }
+
+    /**
+     * If the `Box` component is not imported, this function will create a new import statement for the `Box` component
+     * from `@kuma-ui/core` with the local name `__Box`.
+     */
+    pub fn import_box(&mut self, program: &'a mut Program<'a>) -> &'a mut Program<'a> {
+        let has_box = self.imports.get("Box");
+
+        if has_box.is_none() {
+            let source_literal = StringLiteral {
+                span: SPAN,
+                value: Atom::from("@kuma-ui/core"),
+            };
+
+            let local = BindingIdentifier {
+                span: SPAN,
+                name: Atom::from("__Box"),
+                symbol_id: Default::default(),
+            };
+
+            let identifier_name = IdentifierName {
+                span: SPAN,
+                name: Atom::from("Box"),
+            };
+
+            let imported = ModuleExportName::Identifier(identifier_name);
+
+            let specifier =
+                ImportDeclarationSpecifier::ImportSpecifier(self.ast.alloc(ImportSpecifier {
+                    span: SPAN,
+                    imported,
+                    local,
+                    import_kind: ImportOrExportKind::Value,
+                }));
+
+            let import_declaration = ImportDeclaration {
+                span: SPAN,
+                specifiers: Some(self.ast.new_vec_single(specifier)),
+                source: source_literal,
+                with_clause: None,
+                import_kind: ImportOrExportKind::Value,
+            };
+
+            let module_declaration =
+                ModuleDeclaration::ImportDeclaration(self.ast.alloc(import_declaration));
+
+            let import_statement = Statement::ModuleDeclaration(self.ast.alloc(module_declaration));
+
+            self.imports.insert("Box".to_string(), "__Box".to_string());
+
+            program.body.insert(0, import_statement);
+        }
+
+        program
+    }
+}
+
+pub struct ReplaceKWithBox<'a> {
+    imports: HashMap<String, String>,
+    ast: AstBuilder<'a>,
+}
+
+impl<'a> ReplaceKWithBox<'a> {
+    pub fn new(imports: HashMap<String, String>, allocator: &'a Allocator) -> Self {
+        Self {
+            imports,
+            ast: AstBuilder::new(allocator),
+        }
+    }
+
+    pub fn process_element(
+        &mut self,
+        elem: &mut JSXElement<'a>,
+        k_alias: &str,
+        box_local_name: &str,
+    ) {
+        if let JSXElementName::MemberExpression(member_expr) = &elem.opening_element.name {
+            if let JSXMemberExpressionObject::Identifier(ident) = &member_expr.object {
+                let html_tag = member_expr.property.name.as_str();
+                let html_tag: &'a str = unsafe { std::mem::transmute(html_tag) }; // FIXME: Remove unsafe transmute
```

Comment:
https://github.com/oxc-project/oxc/blob/5081652bc10cdcfd10cf57bf0385ebc7eacd5769/crates/oxc_ast/src/ast_builder.rs#L64-L67

Maybe you want `new_atom` or `String::from_str_in`